### ISSUE TITLE
Excel VBA secure channel support error

### ISSUE THREAD

**Issue by rmanaker** [iexg/IEX-API#872]
## Summary
In working to transition from IEX API to IEX cloud, I am experiencing a secure channel support error

## Details
I have been pulling quotes using MS Excel for over a year using IEX API.  Because of the switch to IEX Cloud, I have been tried to pull data from IEX Cloud.  I am experiencing a secure channel support error "-2147012739 An error occurred in the secure channel support"

### Date/time of issue
15 April 2019

### API URL
https://cloud.iexapis.com/beta

### What you expected to happen
I expected there to be no difference in the APIs

### What actually happened
My code works flawlessly (see below) from the old IEX API (using "https://api.iextrading.com/1.0/stock/apple/quote/latestPrice"), it returns the stock price as expected.  I get the error when I run from the IEX Cloud.   I ran an SSL report on both the old API and the new Cloud site; they appear to have the same SSL configuration.  I assume there is an issue with the token causing some problem with the TLS handshake.  I am a relative novice to VBA and have been unsuccessful in resolving the issue despite a extensive Google search.  Any thoughts?

### Other Information (sample code, links, etc.)
Code:
Function GetStockInfo()

Dim hReq As Object
Dim strUrl As String
strUrl = "https://cloud.iexapis.com/beta/stock/aapl/quote/latestPrice?token=MY_TOKEN" 
  '(MY_TOKEN = secure token from IEX Cloud account)

Set hReq = CreateObject("WinHttp.WinHTTPRequest.5.1")
    With hReq
        .Open "GET", strUrl, False
        .Send
    End With  
Dim strResponse As String
    strResponse = hReq.ResponseText    
Debug.Print strResponse

End Function

**Comment by evancarter-iex** [comment]
@rmanaker I can only go from googling but this seems to cover it: https://stackoverflow.com/questions/44045305/vba-an-error-occurred-in-the-secure-channel-support 
https://www.linkedin.com/pulse/working-vba-tls-protocol-srinath-gudimetla

TL;DR: We are using TLS 1.2 and you need to enable TLS 1.2 support.

**Comment by rmanaker** [comment]
Evan,
I do not have access to edit the registry (business computer).  Also, the IEX API site also uses TLS 1.2 and that does not cause the error.

**Comment by evancarter-iex** [comment]
@rmanaker Hmmm well I double checked our SSL setup between the two sites and they are practically identical. So not sure why this would be causing problems for you, maybe a firewall issue on your businesses side?

**Comment by rmanaker** [comment]
Evan,
Thank you - I ran a check on my side and I am running TLS 1.2 (the TLS version is toward the end):
{"given_cipher_suites":["TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256","TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256","TLS_DHE_RSA_WITH_AES_256_GCM_SHA384","TLS_DHE_RSA_WITH_AES_128_GCM_SHA256","TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384","TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256","TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384","TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256","TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA","TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA","TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA","TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA","TLS_RSA_WITH_AES_256_GCM_SHA384","TLS_RSA_WITH_AES_128_GCM_SHA256","TLS_RSA_WITH_AES_256_CBC_SHA256","TLS_RSA_WITH_AES_128_CBC_SHA256","TLS_RSA_WITH_AES_256_CBC_SHA","TLS_RSA_WITH_AES_128_CBC_SHA","TLS_RSA_WITH_3DES_EDE_CBC_SHA"],"ephemeral_keys_supported":true,"session_ticket_supported":true,"tls_compression_supported":false,"unknown_cipher_suite_supported":false,"beast_vuln":false,"able_to_detect_n_minus_one_splitting":false,"insecure_cipher_suites":{},"tls_version":"TLS 1.2","rating":"Probably Okay"}

**Comment by evancarter-iex** [comment]
@rmanaker What happens when you go to the url in your browser? Also are you using the secret or public token?

**Comment by rmanaker** [comment]
Evan,
I have tried both tokens.  I am unable to get anything in Chrome, IE or Firefox if I paste in the URL.  I even tried downloading the Excel sheet and the Webservice Function does not seem to work.  It is something with the interaction of our server and your cloud site - last night I went to Alpha Advantage which also uses a token or key at the end of the URL and I was able to run code no problem....

**Comment by evancarter-iex** [comment]
@rmanaker When you paste it into Chrome/Firefox, what is the response?

**Comment by rmanaker** [comment]
I get:
Chrome: This site can’t be reached
Firefox: Secure Connection Failed

**Comment by rmanaker** [comment]
I don't know if this helps, but on the HTTP request, if I change from synchronous to asynchronous, get past the initial error in secure channel support but get no data back (the error becomes "-2147483638 The data necessary to complete this operation is not yet available").

**Comment by evancarter-iex** [comment]
@rmanaker what ip are you connecting to our site from?

**Comment by rmanaker** [comment]
**@evancarter-iex**  My ip address is: 214.2.208.246

**Comment by evancarter-iex** [comment]
@rmanaker Hmm you aren't being blocked on ourside. Sounds like a local/business issue with your firewall/connection based on the response of "This Site can't be reached"

**Comment by rmanaker** [comment]
@evancarter-iex  Thank you for your efforts.

**Comment by evancarter-iex** [comment]
@rmanaker Let me know if theres anything else we can do on our end, in the meantime Im going to close this issue
